driver/src/main/java/org/neo4j/driver/v1/Record.java

```diff
@@ -19,33 +19,8 @@
 package org.neo4j.driver.v1;
 
 /**
- * A record is a collection of named fields, and is what makes up the individual items in a {@link
- * Result}
+ * A record is an immutable record view
  */
-public interface Record
-{
-    /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
-     *
-     * @param fieldIndex the field index into the current record
-     * @return the value in the specified field
-     */
-    Value get( int fieldIndex );
-
-    /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
-     * If no value could be found in the specified filed, null will be returned.
-     *
-     * @param fieldName the field to retrieve the value from
-     * @return the value in the specified field or null if no value could be found in the field.
-     */
-    Value get( String fieldName );
-
-    /**
-     * Get an ordered sequence of the field names in this result.
-     *
-     * @return field names
-     */
-    Iterable<String> fieldNames();
-
+public interface Record extends RecordView {
+    // intentionally empty
```

Comment:
I think you look at this from a different end - this is not so much about mutation operations, this is about guarantees when reading.

Assume:

```
interface MapAccess {
  Object value( String key )
}

interface Record extends MapAccess
````
If I now have a `MapAccess m` and then

`````
Object o1 = m.value( "a" );
Object o2 = m.value( "a" );
`````

there is no guarantee for `o1.equals(o2)` (because mutable, by other methods outside of the interface implemented by inheritors).

However if I have a record `Record r` and

`````
Object o1 = r.value( "a" );
Object o2 = r.value( "a" );
`````

then it always holds `o1.equals(o2)`. 

In the end, this is not about what methods are made available, this is about what contract you get when reading. If the inheritance order is flipped, as in `MapAccess extends Record`, then if I had a `Record r = new MapAccess()` my record would not be guaranteed to be immutable (by contract) anymore.